### PR TITLE
[libfaketime] Add new port

### PR DESCRIPTION
**Pull request by dudoslav**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dudoslav** [comment]
@microsoft-github-policy-service agree company="TileDB"

**Comment by Cheney-W** [comment]
```
0.9.10 is declared with 451debe90ea3a82a03d3ba30eeaa0abc3e9409a3, but the local port has a different SHA 43c3d525b4e071c4265c512fda07775af3a0df6a.
Please update the port's version fields and then run:
vcpkg x-add-version libfaketime
git add versions
git commit -m "Update version database"
to add the new version.
To attempt to resolve all errors at once, run:
vcpkg x-add-version --all

```

**Comment by Cheney-W** [comment]
Why don't you use the Makefile provided by the upstream to build this port?

**Comment by dudoslav** [comment]
For our use case only `libfaketime.so` is needed and I prefer using CMake. I hope it is not an issue :) .

**Comment by Cheney-W** [comment]
The vcpkg policy is that if the upstream provides multiple build methods, we prefer CMake. However, if the upstream only provides one available build method, we need to use the upstream's build method instead of using a custom CMake.

File: ports/libfaketime/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+)
+
+vcpkg_cmake_install()
+vcpkg_copy_pdbs()
```
Comment (teo-tsirpanis) [inline ports/libfaketime/portfile.cmake:17]: Try this and if it works, delete `CMakeLists.txt` and the dependency to `vcpkg-cmake`.

```suggestion
vcpkg_install_make()
```

File: ports/libfaketime/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "libfaketime",
+  "version": "0.9.10",
+  "description": "libfaketime modifies the system time for a single application",
+  "homepage": "https://github.com/wolfcw/libfaketime",
+  "license": "GPL-2.0-only",
+  "supports": "linux",
```
Comment (teo-tsirpanis) [inline ports/libfaketime/vcpkg.json:7]:
```suggestion
  "supports": "linux | osx",
```

File: ports/libfaketime/portfile.cmake (hunk 1 above)
Comment (dudoslav) [reply ports/libfaketime/portfile.cmake]: Added

**Comment by dudoslav** [comment]
I reworked this PR to use Make instead of CMake but right now it fails due to I assume incorrect working directory, is there a chance to change it?:
```
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/make V=1 -j 9 -f Makefile all
    Working Directory: /home/dudoslav/Projects/vcpkg/buildtrees/libfaketime/x64-linux-dbg/
    See logs for more information:
      /home/dudoslav/Projects/vcpkg/buildtrees/libfaketime/build-x64-linux-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:136 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:2 (vcpkg_build_make)
  ports/libfaketime/portfile.cmake:10 (vcpkg_install_make)
  scripts/ports.cmake:175 (include)


error: building libfaketime:x64-linux failed with: BUILD_FAILED
Elapsed time to handle libfaketime:x64-linux: 1.7 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libfaketime
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[libfaketime]+Build+error+on+x64-linux&body=Copy+issue+body+from+%2Fhome%2Fdudoslav%2FProjects%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

(base) taco:vcpkg dudoslav$ cat /home/dudoslav/Projects/vcpkg/buildtrees/libfaketime/build-x64-linux-dbg-err.log
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /home/dudoslav/Projects/vcpkg/buildtrees/libfaketime/build-x64-linux-dbg-err.log
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ make: Makefile: No such file or directory
   2   │ make: *** No rule to make target 'Makefile'.  Stop.
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
(base) taco:vcpkg dudoslav$ 
```
1716209136_grim
But I can see that the source is in `src` instead of ` /home/dudoslav/Projects/vcpkg/buildtrees/libfaketime/x64-linux-dbg/`

File: ports/libfaketime/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,17 @@
+vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO wolfcw/libfaketime
+    REF v${VERSION}
+    SHA512 07c431bee21e31343b680d1322dd529ea276e3cc4dbec61646c12bf5d0263163faf6186efeb36b199e24b655578a493c43e3b7a7acf8eba8b9ff84a1e94d618b
+)
+
+vcpkg_configure_make(SOURCE_PATH "${SOURCE_PATH}" SKIP_CONFIGURE)
```
Comment (dg0yt) [inline ports/libfaketime/portfile.cmake:10]:
```suggestion
vcpkg_configure_make(SOURCE_PATH "${SOURCE_PATH}"
    SKIP_CONFIGURE
    COPY_SOURCE
)
```
so that Makefile is copied into the build dir.

File: versions/l-/libfaketime.json
Comment (teo-tsirpanis) [inline versions/l-/libfaketime.json:1]: You should delete this file and revert the changes to `baseline.json`, run `vcpkg x-add-version`, and make and push a commit.

**Comment by dudoslav** [comment]
Sorry for the late reply. Any ideas what might be wrong with these builds?
1717604178_grim

**Comment by teo-tsirpanis** [comment]
BTW @dudoslav the PR has some other errors with regards to the versions database. I cleaned them up and pushed a branch to https://github.com/teo-tsirpanis/vcpkg/tree/libfaketime which you can checkout and force-push. But them I'm getting the same errors. 😕

**Comment by dg0yt** [comment]
IIRC now the key barrier for ports using make without `configure` is passing all required options (per build type) into the `make install` step. Some options are easily available for the `configure` step but not for `make`. 

And that's why I add a small `configure` script to some ports. It creates a `Makefile.vcpkg` which captures the configuration options and passed them on to the actual Makefile. Example: `libcap`.
